requests/utils.py

```diff
@@ -665,6 +665,21 @@ def parse_header_links(value):
 
     return links
 
+def is_valid_location(response):
+    """Verify that multiple Location headers weren't
+    returned from the last response.
+    """
+    if hasattr(response.raw, 'headers'):
+        if hasattr(response.raw.headers, 'getlist'):
+            return len(response.raw.headers.getlist('location')) <= 1
+        else:
+            pass # Do something with header value
```

Comment:
Here we'll likely have a header value similar to `'http://foo.com, http://foo.com'` which could be parsed but probably not reliably. I recognize that URLs can contain unencoded `,` but header values are also defined to be comma separated, so I'm not sure how the case of a multi-value header with values containing commas is handled. A cursory google search wasn't terribly informative. 

Browsers seem to catch this problem at what would be our urllib3 layer before the field merge is ever done. Patching it there is null though, because here we're concerned about how to handle the issue for non-urllib3 responses. Perhaps this is another case we can't handle, but then that puts us back at ground-zero for the pull request (other than preventing the exception for StringIO). Perhaps that what @Lukasa was asking for though and I severely overthought the problem.